packages/web-config-server/src/apiV1/dataBuilders/generic/compose/composeDataByDataPoint.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { fetchComposedData } from '/apiV1/dataBuilders/helpers';
+
+/**
+ * Configuration schema
+ * @typedef {Object} ComposeDataConfig
+ * @property {string} sortOrder
+ * @property {Object<string, { dataBuilder, dataBuilderConfig }>} dataBuilders
+ *
+ * Example config:
+ * {
+ *   "dataBuilders": {
+ *        builder1: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        },
+ *        builder2: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        }
+ *    }
+ * }
+ * 
+ * Data returned from fetchComposedData():
+ * {
+ *     builder1: { 
+ *        data: [{name: dataPoint1: value: '111'}, {name: dataPoint2: value: '222'}}
+ *     },
+ *     builder2: {
+ *        data: [{name: dataPoint1: value: '333'}, {name: dataPoint2: value: '444'}},
+ *     }
+ * }
+ * 
+ * Will be converted and returned:
+ * { 
+ *    data: 
+ *        [
+ *          { name: dataPoint1, builder1: 'ABC', builder2: 'GHI' }, 
+ *          { name: dataPoint2, builder1: 'DEF', builder2: 'JKL' }, 
+ *        ]
+ *    }
+ */
```

Comment:
This is just a reverted version of `composeData` dataBuilder. For this report, we have to return the data like:
```
[
   {
		Pre-School: 0.444,
		Primary: 0.222,
		Secondary: 0.555,
		name: 'Electricity'
   },
   {
		Pre-School: 0.333,
		Primary: 0.444,
		Secondary: 0.555,
		name: 'Functioning smart TV'
	},
]
```

However, it is quite difficult to achieve calculating the percentages for each type of school (stored in attributes column). I guess it is possible, but I can't think of any proper way than adding more config to `percentagesOfValueCounts` data builder which just makes the data builder more complex.

Hence, I went with creating another version of the `composeData` dataBuilder which group the data by the dataPoint (see example documentation in the code)

Reply:
Not sure if this is a good solution but happy for any other suggestions!